src/main/java/com/conveyal/datatools/common/utils/SparkUtils.java

```diff
@@ -99,4 +152,85 @@ public static String formatJobMessage (String jobId, String message) {
             .put("message", message)
             .toString();
     }
+
+    /**
+     * Log Spark requests.
+     */
+    public static void logRequest(Request request, Response response) {
+        logRequestOrResponse(true, request, response);
+    }
+
+    /**
+     * Log Spark responses.
+     */
+    public static void logResponse(Request request, Response response) {
+        logRequestOrResponse(false, request, response);
+    }
+
+    /**
+     * Log request/response.  Pretty print JSON if the content-type is JSON.
+     */
+    public static void logRequestOrResponse(boolean logRequest, Request request, Response response) {
+        // NOTE: Do not attempt to read the body into a string until it has been determined that the content-type is
+        // JSON.
+        HttpServletResponse raw = response.raw();
+        String bodyString = "";
+        try {
+            String contentType;
+            if (logRequest) {
+                contentType = request.contentType();
+            } else {
+                contentType = raw.getHeader("content-type");
+            }
+            if ("application/json".equals(contentType)) {
+                bodyString = logRequest ? request.body() : response.body();
+                if (bodyString != null) {
+                    // Pretty print JSON if ContentType is JSON and body is not empty
+                    JsonNode jsonNode = mapper.readTree(bodyString);
+                    // Add new line for legibility when printing
+                    bodyString = "\n" + mapper.writerWithDefaultPrettyPrinter().writeValueAsString(jsonNode);
+                } else {
+                    bodyString = "{body content is null}";
+                }
+            } else if (contentType != null) {
+                bodyString = String.format("\nnon-JSON body type: %s", contentType);
+            }
+        } catch (IOException e) {
+            LOG.warn("Could not parse JSON", e);
+            bodyString = "\nBad JSON:\n" + bodyString;
+        }
+        logRequestOrResponse(logRequest, request, bodyString, raw.getStatus());
+    }
+
+    public static void logRequestOrResponse(
+        boolean logRequest,
+        Request request,
+        String bodyString,
+        int statusCode
+    ) {
+        Auth0UserProfile userProfile = request.attribute("user");
+        String userEmail = userProfile != null ? userProfile.getEmail() : "no-auth";
+        String queryString = request.queryParams().size() > 0 ? "?" + request.queryString() : "";
+        LOG.info(
+            "{} {} {}: {}{}{}{}",
+            logRequest ? "req" : String.format("res (%s)", statusCode),
+            userEmail,
+            request.requestMethod(),
+            BASE_URL,
+            request.pathInfo(),
+            queryString,
+            trimLines(bodyString)
```

Comment:
I'm beginning to feel that the request/response body (and headers) should only be logged in the debug level. The logs just appear to be too noisy otherwise.

Reply:
In the GraphQL update PR (#118), I was working on refactoring this (see [this code](https://github.com/catalogueglobal/datatools-server/pull/118/files#diff-bcc68337b3f1634cd4453f490fbcbe8dR166-R174)).  How about we tackle that issue there?

Reply:
That direct code link wasn't working for me for some reason, but yea I think handling it there is ok.